frontend/server/server.ts

```diff
@@ -199,48 +221,26 @@ const artifactsHandler = async (req, res) => {
         res.status(500).send('Failed to download GCS file(s). Error: ' + err);
       }
       break;
-    case 'minio':
-      minioClient.getObject(bucket, key, (err, stream) => {
-        if (err) {
-          res.status(500).send(`Failed to get object in bucket ${bucket} at path ${key}: ${err}`);
-          return;
-        }
 
-        try {
-          let contents = '';
-          stream.pipe(new tar.Parse()).on('entry', (entry: Stream) => {
-            entry.on('data', (buffer) => contents += buffer.toString());
-          });
-
-          stream.on('end', () => {
-            res.send(contents);
-          });
-        } catch (err) {
-          res.status(500).send(`Failed to get object in bucket ${bucket} at path ${key}: ${err}`);
-        }
-      });
+    case 'minio':
+      try {
+        res.send(await getTarObjectAsString({bucket, key, client: minioClient}));
```

Comment:
In the next PR, can you please make a function that auto-detects whether the data is gzipped and unpacks if needed?

Reply:
I am thinking whether to change from `node-tar` to `tar-stream` + `maybe-gzip`.

But probably need more testing to ensure it works.